internal/service/common/api/options.go

```diff
@@ -0,0 +1,143 @@
+package api
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+// Fields that can be included/excluded (i.e., optional complex fields where 'complex' means an object or an array)
+const (
+	ExtensionsAttribute   = "extensions"
+	MemberOfAttribute     = "memberOf"
+	CapacityAttribute     = "capacity"
+	CapabilitiesAttribute = "capabilities"
+)
+
+// FieldOptions implements the attribute selection algorithm specified in ETSI GS NFV-SOL 013 V3.5.1, Section 5.3.2.2
+type FieldOptions struct {
+	allFields     bool
+	includeFields []string
+	excludeFields []string
+}
+
+// NewFieldOptions creates a new FieldOptions using the specific values from the API query parameters
+func NewFieldOptions(allFields, fields, excludeFields *string, excludeDefaults ...string) *FieldOptions {
+	result := &FieldOptions{}
+
+	if allFields != nil {
+		if value, err := strconv.ParseBool(*allFields); err == nil {
+			result.allFields = value
+		}
+	}
+
+	if fields != nil {
+		result.includeFields = strings.Split(*fields, ",")
+	}
+
+	if excludeFields != nil {
+		result.excludeFields = strings.Split(*excludeFields, ",")
```

Comment:
after split...should we also call "TrimSpace"...or not needed?

Reply:
good catch!  Fixed.